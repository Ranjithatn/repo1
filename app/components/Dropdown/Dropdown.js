import React from 'react'
import Icon from '../Icon/Icon';

const Dropdown = ({
  text,
  icon,
  children,
  hover = true,
  onClick = () => {},
}) => {
   return (
    <div className={`dropdown ${ hover ? "is-hoverable" : "" }`}   onClick={ e => { onClick && onClick(e)} } >
      <div className="dropdown-trigger">
        <button className="button" aria-haspopup="true" aria-controls="dropdown-menu4">
        { icon && <Icon icon={icon} /> }
          <span>{ text }</span>
          <span className="icon is-small">
            <i className="fa fa-angle-down" aria-hidden="true"></i>
          </span>
        </button>
      </div>
      <div className="dropdown-menu" role="menu" style={{zIndex:55}}>
        <div className="dropdown-content">
          { children }
        </div>
      </div>
    </div>
   )
}

export default Dropdown;
