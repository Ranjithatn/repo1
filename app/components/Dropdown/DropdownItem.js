import React from 'react'
import Icon from '../Icon/Icon';

const DropdownItem = ({
  className,
  text,
  leftIcon,
  rightIcon,
  onClick,
  disabled=false,
}) => {
   return (
    <div className={"dropdown-item" + (className ? " " + className : "")} style={ disabled ? { pointerEvents: "none", color: '#CCC' } : {zIndex:65} } onClick={onClick}>
      { leftIcon && <Icon icon={leftIcon} />}
      { text }
      { rightIcon && <Icon icon={rightIcon} />}
    </div>
   )
}

export default DropdownItem;
