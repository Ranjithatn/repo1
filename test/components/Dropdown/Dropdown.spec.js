import React from "react";
import toJson from "enzyme-to-json";
import Dropdown from "../../../app/components/Dropdown/Dropdown";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe('<Dropdown />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Dropdown />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
