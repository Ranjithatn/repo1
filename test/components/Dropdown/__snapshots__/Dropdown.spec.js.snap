// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should render without crashing 1`] = `
<Dropdown>
  <div
    className="dropdown is-hoverable"
    onClick={[Function]}
  >
    <div
      className="dropdown-trigger"
    >
      <button
        aria-controls="dropdown-menu4"
        aria-haspopup="true"
        className="button"
      >
        <span />
        <span
          className="icon is-small"
        >
          <i
            aria-hidden="true"
            className="fa fa-angle-down"
          />
        </span>
      </button>
    </div>
    <div
      className="dropdown-menu"
      role="menu"
      style={
        Object {
          "zIndex": 55,
        }
      }
    >
      <div
        className="dropdown-content"
      />
    </div>
  </div>
</Dropdown>
`;
